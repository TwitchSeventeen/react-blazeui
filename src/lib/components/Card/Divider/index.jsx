import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Divider (props) {
  const { children, type } = props;
  const classes = classNames('c-card__item', 'c-card__item--divider', { [`c-card__item--${type}`]: type });
  return (
      <div role={type ? null : 'separator'} className={classes}>
        { children }
      </div>
  );
}

Divider.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * The card colouring
   */
  type: PropTypes.oneOf([
    'brand',
    'info',
    'warning',
    'success',
    'error',
  ])
};

Divider.defaultProps = {
  children: null,
  type: null,
};

export default Divider;