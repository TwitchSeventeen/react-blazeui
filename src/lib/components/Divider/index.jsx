import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Divider(props) {
  const {
    children, type, className, ...other
  } = props;
  const classes = classNames('c-divider', { [`c-divider--${type}`]: type }, className);
  return (
    <div {...other} className={classes}>
      <div className="c-divider__content">
        { children }
      </div>
    </div>
  );
}

Divider.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Defines the line style of the divider
   */
  type: PropTypes.oneOf([
    'solid',
    'dashed',
  ]),
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

Divider.defaultProps = {
  children: null,
  type: 'solid',
  className: null,
};

export default Divider;
