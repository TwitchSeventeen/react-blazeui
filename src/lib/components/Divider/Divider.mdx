---
name: Divider
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Divider from './index';

# Divider

Used to separate sections of the page.

## Basic usage
<Playground>
    <Divider>
        Demo
    </Divider>
</Playground>

## Line style
Use the **type** prop to specify the style of line. It can be either _solid_ (default) or _dashed_.
<Playground>
    <Divider type="dashed">
        Demo
    </Divider>
</Playground>

## Properties
<PropsTable of={Divider} />