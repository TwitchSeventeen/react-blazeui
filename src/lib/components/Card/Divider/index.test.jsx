import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Divider from './index';

describe('Card.Divider', () => {
  it('should be defined', () => {
    expect(Divider).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Divider>
        lorem
      </Divider>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a colour', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Divider type="brand">
        lorem
      </Divider>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly as a separator', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Divider separator />,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
