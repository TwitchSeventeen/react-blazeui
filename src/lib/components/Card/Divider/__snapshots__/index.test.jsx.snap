// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.Divider renders correctly 1`] = `
<div
  className="c-card__item c-card__item--divider"
  role="separator"
>
  lorem
</div>
`;

exports[`Card.Divider renders correctly with a colour 1`] = `
<div
  className="c-card__item c-card__item--divider c-card__item--brand"
  role={null}
>
  lorem
</div>
`;
